File: ABC/208/D.py
n, m = map(int, input().split())
inf = float("inf")
dist = [[inf] * n for _ in range(n)]

for i in range(n):
    dist[i][i] = 0

for _ in range(m):
    a, b, c = map(int, input().split())
    a -= 1
    b -= 1
    dist[a][b] = c

res = 0
for i in range(n):
    for j in range(n):
        for k in range(n):
            dist[j][k] = min(dist[j][k], dist[j][i] + dist[i][k])
            if dist[j][k] != inf:
                res += dist[j][k]
print(res)

File: ABC/208/D.cpp
#include <iostream>
#include <vector>
using namespace std;
using ll = long long;

#define rep(i, n) for (int i = 0; i < (n); i++)

int INF = 1001001001;

int main() {
  int n, m;
  cin >> n >> m;
  vector graph(n, vector<int>(n, INF));
  rep(i, n) graph[i][i] = 0;
  rep(i, m) {
    int a, b, c;
    cin >> a >> b >> c;
    a--; b--;
    graph[a][b] = c;
  }

  ll res = 0;
  rep(k, n) {
    rep(i, n) {
      rep(j, n) {
        graph[i][j] = min(graph[i][j], graph[i][k] + graph[k][j]);
        if (graph[i][j] != INF) res += graph[i][j];
      }
    }
  }
  cout << res << endl;
}
